src/tools/rados/rados.cc

```diff
@@ -2855,6 +2855,7 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       goto out;
     }
     io_ctx.set_namespace(all_nspaces);
+    io_ctx.set_osdmap_full_try();
```

Comment:
Unfortunately, from  what I can see using aio_remove() can only use set_osdmap_full_try(), so for now I withdraw my suggestion to support --force-full unless/until another option is added to set that in the io_ctx.